File: plugin/src/main/java/de/s42/tradecycle/service/TradeCycleService.java
package de.s42.tradecycle.service;

import de.s42.core.MessageService;
import de.s42.tradecycle.strategy.KeyboardStrategy;
import de.s42.tradecycle.strategy.ShiftInteractStrategy;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.entity.Villager;
import org.bukkit.event.Listener;

import java.util.Map;

public class TradeCycleService {

    public static final Map<String, Listener> strategies = Map.of(
            "KEYBOARD", new KeyboardStrategy(),
            "SHIFT_INTERACT", new ShiftInteractStrategy()
    );

    private final MessageService messageService;
    private final FileConfiguration config;

    public TradeCycleService(MessageService messageService, FileConfiguration config) {
        this.messageService = messageService;
        this.config = config;
    }

    public void sendActionBar(Player player, String messageKey) {
        String message = config.getString("messages." + messageKey);
        if (message == null) {
            throw new IllegalArgumentException("Message key '" + messageKey + "' not found in config");
        }
        messageService.sendActionBar(player, message);
    }

    public void cycleTrade(Villager villager) {
        Villager.Profession profession = villager.getProfession();
        villager.setProfession(Villager.Profession.NONE);
        villager.setProfession(profession);
    }

    public boolean isLocked(Villager villager) {
        return villager.getVillagerExperience() > 0;
    }

    public boolean hasProfession(Villager villager) {
        return villager.getProfession() != Villager.Profession.NONE;
    }
}

File: plugin/src/main/kotlin/tr/s42/tradecycle/service/TradeCycleService.kt
package tr.s42.tradecycle.service

import tr.s42.tradecycle.strategy.KeyboardStrategy
import tr.s42.tradecycle.strategy.ShiftInteractStrategy
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.entity.Player
import org.bukkit.entity.Villager
import org.bukkit.event.Listener
import tr.s42.core.MessageService

class TradeCycleService(
    private val messageService: MessageService,
    private val config: FileConfiguration
) {

    companion object {
        val strategies: Map<String, Listener> = mapOf(
            "KEYBOARD" to KeyboardStrategy(),
            "SHIFT_INTERACT" to ShiftInteractStrategy()
        )
    }

    fun sendActionBar(player: Player, messageKey: String) {
        val message: String = config.getString("messages.$messageKey")
            ?: throw IllegalArgumentException("Message key '$messageKey' not found in config")
        messageService.sendActionBar(player, message)
    }

    fun cycleTrade(villager: Villager) {
        val profession = villager.profession
        villager.profession = Villager.Profession.NONE
        villager.profession = profession
    }

    fun isLocked(villager: Villager): Boolean {
        return villager.villagerExperience > 0
    }

    fun hasProfession(villager: Villager): Boolean {
        return villager.profession != Villager.Profession.NONE
    }

}